config/configcompression/compressiontype.go

```diff
@@ -40,3 +53,28 @@ func (ct *Type) UnmarshalText(in []byte) error {
 	}
 	return fmt.Errorf("unsupported compression type %q", typ)
 }
+
+func (cc *CompressionConfig) Validate() error {
+	if (typ == TypeGzip && isValidLevel(int(cc.Level))) ||
+		(typ == TypeZlib && isValidLevel(int(cc.Level))) ||
+		(typ == TypeDeflate && isValidLevel(int(cc.Level))) ||
+		typ == TypeSnappy ||
+		typ == TypeLz4 ||
+		typ == TypeZstd ||
+		typ == typeNone ||
+		typ == typeEmpty {
+		return nil
+	}
+
+	return fmt.Errorf("unsupported compression type and level %s - %d", typ, cc.Level)
+}
+
+// Checks the validity of zlib/gzip/flate compression levels
+func isValidLevel(level int) bool {
```

Comment:
Why don't we accept `Level` instead of casting it back and forth?
```suggestion
func isValidLevel(level Level) bool {
```